### ISSUE TITLE
ssh-key failed to authenticate

### ISSUE THREAD

**Issue by jpnadas** [mssun/passforios#260]
Hi,

I am having trouble with authenticating my ssh-key and I am getting the error: 

Failed to authenticate SSH session: Unable to extract public key from private key file: Wrong passphrase or invalid/unrecognized private key file format.

I generated the ssh key on my laptop via `ssh-keygen` and added it to the app via the qrcode scanner.

The way I generated the qrcode was `cat ~/.ssh/id_rsa | qrencode -o code.png`

When I attempt to clone the repo I get the aforementioned error.

I am sure the passphrase is correct. I even attempted to create a new key with a new passphrase and got the same result.

**Comment by yishilin14** [comment]
As far as I could remember, I tried to encode my SSH keys in a single QR code, my camera was not able to scan it.

You can try this simple tool to generate qr codes: https://github.com/yishilin14/asc-key-to-qr-code-gif. It works for both PGP keys and SSH keys. It generate a gif with two pictures in it given my SSH key.

**Comment by jpnadas** [comment]
Thanks for the suggestion. I had seen this option, but I could not find it on the AUR and I prefer to avoid compiling from source when possible.

I will give it a go anyways just to see if that solves the issue.

Cheers.

**Comment by yishilin14** [comment]
It's just one [shell script](https://github.com/yishilin14/asc-key-to-qr-code-gif/blob/master/asc-to-gif.sh). I guess you could find all three common dependencies on the AUR.

Basically, it splits your key file into two or more pieces and generates one png for each split. Then you could use Pass to scan those png in sequence. You could do all these manually.

**Comment by jpnadas** [comment]
I have generated the gif with the asc-key-to-qr-code-gif script as suggested, but the result is the same.

In fact, scanning the image generated from `cat ~/.ssh/id_rsa | qrencode -o code.png` adds the same infomration to the ASCII-ARMOR ENCRYPTED PRIVATE KEY field on the app.

In fact, I tried creating a new ssh key without password and the result is still the same.

I don't think the issue is the qrcode.
